src/main/scala/is/hail/keytable/KeyTable.scala

```diff
@@ -225,21 +225,21 @@ class KeyTable(val hc: HailContext,
            |""".stripMargin)
       false
     } else {
-      val localColumns = columns
-      val localOtherColumns = other.columns
 
       keyedRDD().groupByKey().fullOuterJoin(other.keyedRDD().groupByKey()).forall { case (k, (v1, v2)) =>
         (v1, v2) match {
           case (None, None) => true
           case (Some(x), Some(y)) =>
-            val r1 = x.map(r => localColumns.zip(r.toSeq).toMap).toSet
-            val r2 = y.map(r => localOtherColumns.zip(r.toSeq).toMap).toSet
-            val res = r1 == r2
+            val r1 = x.toArray
+            val r2 = y.toArray
+            val res = if (r1.length != r2.length)
+              false
+            else r1.counter() == r2.counter()
```

Comment:
Nice, I agree this should have the same behavior as the original and should be more efficient. Namely, two key tables are the same if they have:

1) the same schema (as struct => enforcing column order)
2) the same keys (as array => enforcing key order)
3) the same set of rows, with the same multiplicities as rows need not be unique (not enforcing order)

On 2, do we enforce order so that field order of joined key table is well-defined? Or are there other reasons?

Reply:
We do enforce the order on 2, because we join two tables based on number of keys and types, which lets you join something keyed by `["Sample"]` with something keyed by `["s"]`, but could cause unexpected behavior if you want to join something that's keyed by `["key1", "key2"]` with something keyed by `["key2", key1"]` (if key1 and key2 are the same type)